### ISSUE TITLE
HATS implementation double check

### ISSUE THREAD

**Issue by biphasic** [neuromorphs/tonic#151]
Compare current implementation with others, e.g. https://github.com/rfma23/HATS

**Comment by fabhertz95** [comment]
Hi everyone!
I was trying to compare my implementation of HATS with yours and I noticed something strange in your code. In particular, in ```functional/to_averaged_timesurface.py``` you write at line 56:

```
all_surfaces = np.zeros((n_of_events, n_of_pols, surface_size, surface_size))
```

From what I understand, you do not return an array of histograms equal to the number of cells in which the frame is divided, but you simply return the accumulated time surfaces corresponding to each event in the stream. This does not seem right to me, since the output of an HATS encoder should be an array of histograms. For this reason, I have not been able to train a classifier of the data provided by tonic. 

I expected a data structure such as:

```
all_surfaces = np.zeros((n_of_cells, n_of_pols, surface_size, surface_size))
```
Like in the following image from the original article:
![image](https://user-images.githubusercontent.com/45739782/157253399-f41427c9-2fff-43ca-98db-b5fd85704592.png)
Moreover, at line 89:
```
timesurface[radius, radius] = 1
```
It seems almost like you are including the event for which you are computing the time surface in the computation, by computing ```timesurface[radius, radius] = np.exp((t_i-t_i)/tau) == np.exp(0) == 1```, which should not be done, as it is shown in the pseudocode from the original article: the event is added to the cell only _after_ the time surface has been computed.
![image](https://user-images.githubusercontent.com/45739782/157253559-bee824f6-ba05-46c0-bf48-f8a5e990c613.png)
Am I missing something? In the documentation it seems that the desired result is having the time surfaces accumulated for each event in the stream. I am a newbie in the field :)
Thank you in advance for your help!

**Comment by biphasic** [comment]
fixed in #184
